docs/_includes/Statements.md

```diff
@@ -644,6 +640,42 @@ An `Assume` statement cannot be compiled. In fact, the compiler
 will complain if it finds an **assume** anywhere where it has not
 been replaced through a refinement steop.
 
+Using `...` as the argument of the statement is part of module refinement, as described [here](#sec-module-refinement).
+
+## Expect Statement
+
+````
+ExpectStmt =
+    "expect" { Attribute }
+    ( Expression(allowLemma: false, allowLambda: true)
+    | "..."
+    )
+    [ "," Expression(allowLemma: false, allowLambda: true) ]
+    ";"
+````
+
+The `expect` statement states a boolean expression that is checked to be true
+at run-time. That is, the compiler inserts into the run-time executable a 
+check that the given expression is true; if the expression is false, then
+the execution of the program halts immediately. If a second argument is
+given, it is a string that is included in the message emitted by the program
+when it halts; otherwise a default message is emitted.
+
+Because the expect expression is compiled, it cannot be a ghost expression.
+
+The `expect` statement is analogous to the `assert` statement. The `assert` expression
+is checked by the verifier and ignored at runtime; an `expect` expression is checked
+at runtime and ignored by the verifier.  Paired `assert` and `expect` statements checking the
+same expression can be used to do runtime checking before there is a successful proof
+of an assert statement (or to help debug one that is unprovable).[^expect]
```

Comment:
I find this confusing. If there's not a successful proof, then the verifier will complain about the `assert` and (by default) you won't get compiled code. I think it's better to say that there are two intended use cases:

* Check assumptions about external code (see `{:extern}` attribute), where there is no other good recourse if the assumption does not hold 
* Check expected results in unit tests (see `{:test}` attribute)